src/coreclr/src/vm/jithelpers.cpp

```diff
@@ -5041,6 +5041,59 @@ Thread * __stdcall JIT_InitPInvokeFrame(InlinedCallFrame *pFrame, PTR_VOID StubS
 EXTERN_C void JIT_PInvokeBegin(InlinedCallFrame* pFrame);
 EXTERN_C void JIT_PInvokeEnd(InlinedCallFrame* pFrame);
 
+// Forward declaration
+EXTERN_C void STDCALL ReversePInvokeBadTransition();
+
+// This is a slower version of the reverse PInvoke enter function.
+NOINLINE static void JIT_ReversePInvokeEnterRare(ReversePInvokeFrame* frame)
+{
+    _ASSERTE(frame != NULL);
+
+    Thread* thread = GetThreadNULLOk();
+    if (thread == NULL)
+        thread = CreateThreadBlockThrow();
```

Comment:
If the exception thrown from CreateThreadBlockThrow going to show up with the right diagnostic? 

I think we may need

```
    INSTALL_MANAGED_EXCEPTION_DISPATCHER
    INSTALL_UNWIND_AND_CONTINUE_HANDLER
```

around `CreateThreadBlockThrow` to make it work properly.

Reply:
Unfortunately can't do that since the `INSTALL_UNWIND_AND_CONTINUE_HANDLER` requires `CURRENT_THREAD` to be non-null and we are trying to address that here. I will update this to handle the case in a fail fast manner though.